test_communication/test/test_timeout_subscriber.cpp

```diff
@@ -0,0 +1,156 @@
+}
+
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+  if (argc != 2) {
+    fprintf(stderr, "Wrong number of arguments, pass one message type\n");
+    return 1;
+  }
+
+  auto start = std::chrono::steady_clock::now();
+
+  std::string message = argv[1];
+  auto node = rclcpp::Node::make_shared(std::string("test_subscriber_") + message);
+
+  auto messages_empty = get_messages_empty();
+  auto messages_primitives = get_messages_primitives();
+  auto messages_static_array_primitives = get_messages_static_array_primitives();
+  auto messages_dynamic_array_primitives = get_messages_dynamic_array_primitives();
+  auto messages_nested = get_messages_nested();
+  auto messages_dynamic_array_nested = get_messages_dynamic_array_nested();
+  auto messages_static_array_nested = get_messages_static_array_nested();
+  auto messages_builtins = get_messages_builtins();
+  rclcpp::executors::SingleThreadedExecutor executor;
+  //executor.add_node(node);
+
+  rclcpp::subscription::SubscriptionBase::SharedPtr subscriber;
+  std::vector<bool> received_messages;  // collect flags about received messages
+  if (message == "empty") {
+    subscriber = subscribe<test_communication::msg::Empty>(
+      node, message, messages_empty, received_messages);
+  } else if (message == "primitives") {
+    subscriber = subscribe<test_communication::msg::Primitives>(
+      node, message, messages_primitives, received_messages);
+  } else if (message == "staticarrayprimitives") {
+    subscriber = subscribe<test_communication::msg::StaticArrayPrimitives>(
+      node, message, messages_static_array_primitives, received_messages);
+  } else if (message == "dynamicarrayprimitives") {
+    subscriber = subscribe<test_communication::msg::DynamicArrayPrimitives>(
+      node, message, messages_dynamic_array_primitives, received_messages);
+  } else if (message == "nested") {
+    subscriber = subscribe<test_communication::msg::Nested>(
+      node, message, messages_nested, received_messages);
+  } else if (message == "dynamicarraynested") {
+    subscriber = subscribe<test_communication::msg::DynamicArrayNested>(
+      node, message, messages_dynamic_array_nested, received_messages);
+  } else if (message == "staticarraynested") {
+    subscriber = subscribe<test_communication::msg::StaticArrayNested>(
+      node, message, messages_static_array_nested, received_messages);
+  } else if (message == "builtins") {
+    subscriber = subscribe<test_communication::msg::Builtins>(
+      node, message, messages_builtins, received_messages);
+  } else {
+    fprintf(stderr, "Unknown message argument '%s'\n", message.c_str());
+    return 1;
+  }
+
+  size_t num_cycles = 5; //TODO: args
+  auto blocking_timeout = std::chrono::milliseconds(100);
+  auto tolerance = std::chrono::milliseconds(10);
+  for (size_t i = 0; i < num_cycles; ++i) {
+    auto nonblocking_start = std::chrono::steady_clock::now();
+    executor.spin_node_once(node, std::chrono::milliseconds::zero());
```

Comment:
Is this `spin_node_once` call supposed to sleep for 100 ms when the timeout parameter is zero?